src/diagnostics.jl

```diff
@@ -84,6 +84,12 @@ function stop_tracking_time()
     compute_context[] = nothing
 end
 
+struct TrackTime
+    obj
```

Comment:
This is actually an NDSparse object -- this is quite useful to filter timings for specific steps (e.g. comm / compute) or specific processes. It would be nice to make a note of how to get to this in the printing? Once taken out, the NDSparse has to be wrapped again in this type to display as it does...